### ISSUE TITLE
Deploy meteor 1.3 on heroku keep failing

### ISSUE THREAD

**Issue by xiao25** [AdmitHub/meteor-buildpack-horse#84]
```
ztx$ heroku config:set ROOT_URL=https://guarded-coast-20468.herokuapp.com
Setting config vars and restarting guarded-coast-20468... done
ROOT_URL: https://guarded-coast-20468.herokuapp.com
wirelessprvnat-172-17-208-227:EntityRelationSearch ztx$ git push heroku master
Counting objects: 93, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (84/84), done.
Writing objects: 100% (93/93), 15.37 KiB | 0 bytes/s, done.
Total 93 (delta 35), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Fetching set buildpack https://github.com/lirbank/meteor-buildpack-horse.git... done
remote: -----> Node.js app detected
remote: -----> Installing node
remote: -----> Installing meteor
remote: Downloading Meteor distribution
remote: ######################################################################## 100.0%
remote: 
remote: Meteor 1.3.1 has been installed in your home directory (~/.meteor).
remote: 
remote: Now you need to do one of the following:
remote: 
remote:   (1) Add "$HOME/.meteor" to your path, or
remote:   (2) Run this command as root:
remote:         cp "/tmp/buildpack20160409-162-14b3lkd/meteor-OYKc/.meteor/packages/meteor-tool/1.3.1/mt-os.linux.x86_64/scripts/admin/launch-meteor" /usr/bin/meteor
remote: 
remote: Then to get started, take a look at 'meteor --help' or see the docs at
remote: docs.meteor.com.
remote: -----> Bundling bundle
remote: 
remote: 
remote: While removing platforms:
remote: error: ios: platform is not in this project
remote: error: android: platform is not in this project
remote: 
remote: 
remote:  !     Push rejected, failed to compile Node.js app
remote: 
remote: Verifying deploy...
remote: 
remote: !	Push rejected to guarded-coast-20468.
remote: 
To https://git.heroku.com/guarded-coast-20468.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/guarded-coast-20468.git'
```

Note: I tried run 
heroku config:set BUILDPACK_URL=https://github.com/AdmitHub/meteor-buildpack-horse.git
heroku addons:add mongolab
before I run the code before

**Comment by xiao25** [comment]
Problem solved by adding yours platform on my project, although I don't need it....

**Comment by yourcelf** [comment]
Which platform did you need to add?

**Comment by gg2001** [comment]
Yes, I am very curious what platform you added. I am getting the exact same issue here.

**Comment by mrlowe** [comment]
I added the android and ios platforms and it built. That's not an issue for me because I'll be adding those platforms to this project anyway, but other people might be interested in figuring out exactly what's going on here. =/

**Comment by yourcelf** [comment]
@mrlowe, @gg2001, and @xiao25: which versions of Meteor are you targeting?  And are you sure that the error you're encountering is due to the presence/lack of ios/Android platforms, and not something else?  Are you running with the BUILDPACK_PRELAUNCH_METEOR flag?

This buildpack removes those platforms for meteor versions < 1.3 that won't build on the server if required platforms are missing.  This has always printed an "error:..." to the console during build if they weren't present, but it was not a blocking error (e.g. the calls return with a "success" status code even though they print an error).  If Meteor has changed to return an error status code from those calls, we'll need to make changes.

The logic is different when running with BUILDPACK_PRELAUNCH_METEOR, so I just wanted to check and see if the problem lies there (you shouldn't be running with that flag unless you know you need to prelaunch the meteor server process to build assets before bundling).

**Comment by mrlowe** [comment]
@yourcelf Nope, not using BUILDPACK_PRELAUNCH_METEOR. Targeting 1.3.1. The app deployed fine to heroku until my latest "meteor update" (from 1.2 to 1.3.1).

**Comment by gg2001** [comment]
Not using BUILDPACK_PRELAUNCH_METEOR. Will try updating.

**Comment by yourcelf** [comment]
Meteor 1.3.1 changed the way it handles mobile platforms, and also added a --server-only flag which allows us to build on the server without removing them.  This should now be fixed in #90 -- we know check whether the version of meteor your app uses supports `--server-only`, and fall back to the old platform removal strategy if so.  Please reopen if you experience further problems.

**Comment by mrlowe** [comment]
Very nice, thanks!
